code/modules/network/network.dm

```diff
@@ -0,0 +1,119 @@
+/*
+	Upgraded devices, such as borgs, the AI, or "Hacktools" have several notable features:
+	they are able to see stealthed networks.
+	they are able to pass "-b" as an argument in place of an encryption key to make an attempt at brute-forcing the network. This will take some time, and may trigger the networks security system, but will eventually get you in.
+*/
+
+var/global/list/networks_by_id = list()
+var/global/list/datum/network/networks_by_wide = list()
+var/global/list/active_network_ids = list()
+
+proc/clean_network_command(command = "")
+	var/list/butchered = explode_text(command, " -") // allows for arguments like {text1 = "hello "} {text2 = "world"} to be caused when executing something like {print -text1 = "hello " -text2 = "world"}
+	command = butchered[1]
+	var/list/arguments = list()
+	for(var/i=2, i<=butchered.len, i++)
+		arguments[i-1] = butchered[i]
+	return list(command, arguments)
+
+/datum/network_argument
+	var/main_command
+	var/list/args = list()
+
+/datum/network_argument/New(var/command)
+	var/list/clean_command = clean_network_command(command)
+	main_command = clean_command[1]
+	args = clean_command[2]
+
+/datum/network
+	var/atom/holder = null							// The holder (atom that contains this network datum).
+	var/id = null									// A unique id
+	var/password = null								// A password to prevent unauthorized access to the network.
+	var/stealth = 0									// Whether the network should be hidden from standard probing and prying.
+	var/wireless = 0								// Whether the network can be accessed remotely.
+	var/datum/network/root 							// The root network, if this is a subnetwork.
+	var/list/datum/network/sub = list() 			// A list of networks that are a subnetwork of this one.
+	var/list/datum/network_command/commands = list()	// A list of commands this network has access to. Works kinda like virus symptoms.
+
+/datum/network/New(atom/holder)
+	..()
+	src.holder = holder
+	if(id)
+		while(networks_by_id[id])
```

Comment:
instead of this nonsense, you can just make ```id``` ```var/static/id``` and ```id++``` before using the ID, this ensures a unique (although linearly numeric) ID.